### PR TITLE
Refactor persistent volume & volume snapshot contents syncers

### PR DESCRIPTION
**Pull request by FabianKramm**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

**Please provide a short message that should be published in the vcluster release notes**
Part of ENG-4168

### CODE REVIEW

File: pkg/controllers/resources/persistentvolumes/syncer.go [hunk 1]
```diff
@@ -192,16 +163,43 @@ func (s *persistentVolumeSyncer) Sync(ctx *synccontext.SyncContext, pObj client.
...
+	targetObject.Spec.MountOptions = sourceObject.Spec.MountOptions
+
+	// update virtual object
+	err = s.translateUpdateBackwards(ctx, vPersistentVolume, pPersistentVolume, vPvc)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// update virtual status
+	vPersistentVolume.Status = *pPersistentVolume.Status.DeepCopy()
```
Comment (facchettos) [inline pkg/controllers/resources/persistentvolumes/syncer.go:196]: why do we do a deepcopy here?

Comment (FabianKramm) [reply pkg/controllers/resources/persistentvolumes/syncer.go:196]: Yeah I guess its not necessary I can remove that if you want

File: pkg/controllers/resources/volumesnapshots/volumesnapshotcontents/syncer.go [hunk 2]
```diff
@@ -177,16 +164,30 @@ func (s *volumeSnapshotContentSyncer) Sync(ctx *synccontext.SyncContext, pObj cl
...
+	}()
+
+	// update virtual object
+	s.translateUpdateBackwards(pVSC, vVSC)
+
+	// update virtual status
+	vVSC.Status = pVSC.Status.DeepCopy()
+
+	// update host object
+	if vVSC.Annotations == nil || vVSC.Annotations[constants.HostClusterVSCAnnotation] == "" {
```
Comment (facchettos) [inline pkg/controllers/resources/volumesnapshots/volumesnapshotcontents/syncer.go:187]:
```suggestion
	if vVSC.Annotations[constants.HostClusterVSCAnnotation] == "" {
```
would be equivalent I think

File: pkg/controllers/resources/persistentvolumes/syncer.go [hunk 3]
```diff
@@ -192,16 +163,43 @@ func (s *persistentVolumeSyncer) Sync(ctx *synccontext.SyncContext, pObj client.
...
+	err = s.translateUpdateBackwards(ctx, vPersistentVolume, pPersistentVolume, vPvc)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// update virtual status
+	vPersistentVolume.Status = pPersistentVolume.Status
+
+	// update host object
+	if vPersistentVolume.Annotations == nil || vPersistentVolume.Annotations[constants.HostClusterPersistentVolumeAnnotation] == "" {
```
Comment (facchettos) [inline pkg/controllers/resources/persistentvolumes/syncer.go:199]:
```suggestion
	if vPersistentVolume.Annotations[constants.HostClusterPersistentVolumeAnnotation] == "" {
```

**Review by facchettos — APPROVED** [review]
little nits for map access but otherwise lgtm

**Pull request merged**